commons/src/main/java/org/infinispan/commons/configuration/ClassWhiteList.java

```diff
@@ -45,23 +45,49 @@
 
    static {
       // Classes always allowed
+      // Primitives
+      SYS_ALLOWED_CLASSES.add(byte.class.getName());
+      SYS_ALLOWED_CLASSES.add(short.class.getName());
+      SYS_ALLOWED_CLASSES.add(int.class.getName());
+      SYS_ALLOWED_CLASSES.add(long.class.getName());
+      SYS_ALLOWED_CLASSES.add(float.class.getName());
+      SYS_ALLOWED_CLASSES.add(double.class.getName());
+      SYS_ALLOWED_CLASSES.add(char.class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean.class.getName());
+
+      // Primitive Arrays
+      SYS_ALLOWED_CLASSES.add(byte[].class.getName());
+      SYS_ALLOWED_CLASSES.add(short[].class.getName());
+      SYS_ALLOWED_CLASSES.add(int[].class.getName());
+      SYS_ALLOWED_CLASSES.add(long[].class.getName());
+      SYS_ALLOWED_CLASSES.add(float[].class.getName());
+      SYS_ALLOWED_CLASSES.add(double[].class.getName());
+      SYS_ALLOWED_CLASSES.add(char[].class.getName());
+      SYS_ALLOWED_CLASSES.add(boolean[].class.getName());
+
+      // Boxed Primitives
       SYS_ALLOWED_CLASSES.add(Byte.class.getName());
       SYS_ALLOWED_CLASSES.add(Short.class.getName());
       SYS_ALLOWED_CLASSES.add(Integer.class.getName());
       SYS_ALLOWED_CLASSES.add(Long.class.getName());
       SYS_ALLOWED_CLASSES.add(Float.class.getName());
       SYS_ALLOWED_CLASSES.add(Double.class.getName());
-      SYS_ALLOWED_CLASSES.add(Number.class.getName());
       SYS_ALLOWED_CLASSES.add(Character.class.getName());
       SYS_ALLOWED_CLASSES.add(String.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashSet.class.getName());
-      SYS_ALLOWED_CLASSES.add(HashMap.class.getName());
-      SYS_ALLOWED_CLASSES.add(Date.class.getName());
-      SYS_ALLOWED_CLASSES.add(Enum.class.getName());
       SYS_ALLOWED_CLASSES.add(Boolean.class.getName());
+
+      // Util
       SYS_ALLOWED_CLASSES.add(ArrayList.class.getName());
-      SYS_ALLOWED_CLASSES.add(Object[].class.getName());
-      SYS_ALLOWED_CLASSES.add("org.infinispan.jboss.marshalling.commons.JBossExternalizerAdapter");
+      SYS_ALLOWED_CLASSES.add(Date.class.getName());
```

Comment:
What about Instant and other java time classes ?

Reply:
Do you think we should include all of `java.time.*`? Or just `Instant` and a few select others?

Reply:
Instant should probably be enough for now. There are too many odd things in java time ...

Reply:
Ok, Instant has been added.